pageserver/src/tenant.rs

```diff
@@ -1446,6 +1476,20 @@ impl Tenant {
 
         Ok(())
     }
+
+    /// Gathers inputs from all of the timelines to produce a pricing model input.
+    ///
+    /// Future is cancellation safe. Only one calculation can be running at once per tenant.
+    #[instrument(skip_all, fields(tenant_id=%self.tenant_id))]
+    pub async fn gather_pricing_inputs(&self) -> anyhow::Result<pricing::ModelInputs> {
+        use pricing::gather_inputs;
+
+        let logical_sizes_at_once = self.conf.concurrent_pricing_logical_size_queries.inner();
+
+        let mut shared_cache = self.cached_logical_sizes.lock().await;
```

Comment:
That leaves a bad taste, since presumably every other HTTP request has to wait on other (potentially lengthy) calculation to finish.

Again, feels like incremental size computation could help with that.

Reply:
I'll record a `TODO:` above this, once I find your other comment about the incremental computation.

Reply:
In next push, waiting for the e2e test run.